android/src/main/kotlin/com/mr/flutter/plugin/filepicker/FileUtils.kt

```diff
@@ -159,15 +159,16 @@ object FileUtils {
                     intent.putExtra(Intent.EXTRA_MIME_TYPES, allowedExtensions)
                 }
             } else {
-                intent = Intent(Intent.ACTION_OPEN_DOCUMENT).apply {
+                intent = Intent(Intent.ACTION_GET_CONTENT).apply {
                     addCategory(Intent.CATEGORY_OPENABLE)
                     type = this@startFileExplorer.type
+                    if (!allowedExtensions.isNullOrEmpty()) {
+                        putExtra(Intent.EXTRA_MIME_TYPES, allowedExtensions!!.toTypedArray())
+                    }else{
```

Comment:
Small formatting nit.

```suggestion
                    } else {
```